internal/rbd/nodeserver.go

```diff
@@ -241,8 +241,12 @@ func populateRbdVol(
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if req.GetVolumeContext()["mounter"] == rbdDefaultMounter &&
-		!isKrbdFeatureSupported(ctx, strings.Join(rv.ImageFeatureSet.Names(), ",")) {
+	isFeatureExist, err := isKrbdFeatureSupported(ctx, strings.Join(rv.ImageFeatureSet.Names(), ","))
+	if errors.Is(err, os.ErrNotExist) {
+		// do not fallback if /sys/bus/rbd/supported_features is absent
+		isFeatureExist = true
```

Comment:
so, if the file is missing, assume the feature is present? Is that not problematic with new features on old RBD drivers?

Reply:
We do not have a way to find what features are present when `/sys/bus/rbd/supported_features` is absent, indeed that is the reason why `/sys/bus/rbd/supported_features` file was introduced.

Are you suggesting that we should fallback to NBD if the `/sys/bus/rbd/supported_features` is absent?

Reply:
Just wondering what the most appropriate default would be. If `/sys/bus/rbd/supported_features` is missing, the driver is old, so fewer features would be supported. Consider having a list of supported features of the last driver version that does not provide this API.

Reply:
@Madhu-1 what is your take on this? Instead should we fallback to NBD incase if the krbd driver is old then?

Reply:
but it's not expected behavior for existing clusters.

Reply:
Then I suppose this will be a breaking change and should be introduced in 3.7.0 only. Thoughts?

Reply:
IMO this will be good only if nbd is stable and can be used in production for 3.7.0.

Reply:
do you know what features are supported by the most recent driver that does not have the `/sys/bus/rbd/supported_features` file? In case the file is missing, you could check if it is a feature that is supported long (enough).

Reply:
Upstream /sys/bus/rbd/supported_features is part of Linux kernel v4.11.0
RBD_FEATURE_LAYERING is added around Linux v3.6,
RBD_FEATURE_STRIPINGV2 is added around v3.9,
RBD_FEATURE_EXCLUSIVE_LOCK is added around v4.8, and
RBD_FEATURE_DATA_POOL around v4.10

Reply:
The reporter of the issue used Linux 4.4.0. I guess some Debian version might still be on such an old kernel?

As you have the features and their kernel versions, you can use that (and ignore backports). In case `/sys/bus/rbd/supported_features` exists, use that as authoritative source.

Reply:
@Madhu-1 Are you okay with ignoring backports too and basically with Niels comment above?

Reply:
Yes, should be fine :+1:

Reply:
Waiting on an update for this, marking as required change.

Reply:
Done now, PTAL.